Intervention V&V

collijk opened the pull request:
## Intervention V&V
<!-- Ideally, <=50 chars. 50 chars is here..: -->

### Description
<!-- For use in commit message, wrap at 72 chars. 72 chars is here: -->
*Category*: V&V
*JIRA issue*: [MIC-3161](https://jira.ihme.washington.edu/browse/MIC-3161)

- Address Ali's V&V issues
- Add some timing 
- Do some optimization in the observers


### Verification and Testing
Ali's V&V notebooks + timing info for the optimizations.

Commit:
Fix baseline IFA effect

Commit:
Fix anemia ylds in postpartum

Commit:
Fix timing of interventions

Commit:
Stratify anemia by interventions

Commit:
Update results stratification for anemia to include treatment

Commit:
Add back age stratification

Commit:
Fix naming

Commit:
Add bmi stratification for hemoglobin

Commit:
Speed up observers

Commit:
Reduce a little further

Commit:
Revert artifact path

mattkappel commented on src/vivarium_gates_iv_iron/constants/results.py:
```diff
@@ -30,9 +30,9 @@
+DEATH_COLUMN_TEMPLATE = "death_due_to_{CAUSE_OF_DEATH}_year_{YEAR}_pregnancy_status_{PREGNANCY_STATE}_maternal_supplementation_{SUPPLEMENTATION}_antenatal_iv_iron_{ANTENATAL_IV_IRON}_postpartum_iv_iron_{POSTPARTUM_IV_IRON}"
+YLLS_COLUMN_TEMPLATE = "ylls_due_to_{CAUSE_OF_DEATH}_year_{YEAR}_pregnancy_status_{PREGNANCY_STATE}_maternal_supplementation_{SUPPLEMENTATION}_antenatal_iv_iron_{ANTENATAL_IV_IRON}_postpartum_iv_iron_{POSTPARTUM_IV_IRON}"
+YLDS_COLUMN_TEMPLATE = "ylds_due_to_{CAUSE_OF_DISABILITY}_year_{YEAR}_pregnancy_status_{PREGNANCY_STATE}_maternal_supplementation_{SUPPLEMENTATION}_antenatal_iv_iron_{ANTENATAL_IV_IRON}_postpartum_iv_iron_{POSTPARTUM_IV_IRON}"
 PREGNANCY_OUTCOME_COUNT_COLUMN_TEMPLATE = "{PREGNANCY_OUTCOME}_count_year_{YEAR}"
```
These templates are missing `_age_{AGE_GROUP}`. Did `make_results` work?

Commit:
Birth recorder interventions (#67)

*Category*: feature
*JIRA issue*: [MIC-3115](https://jira.ihme.washington.edu/browse/MIC-3115)

Add maternal interventions to birth recorder

Verification and Testing
Runs, then usage of outputs in child model
